### DISCUSSION TITLE
Change autofucus with livewire

### DISCUSSION THREAD

**Discussion by mldgato** [livewire/livewire#4251] [1. Help]
I have a problem in a crud, I need that when the page loads the autofocus it is in the search engine so that the user does not have to click on the input, that works well because in said input I have placed the autofocus property, then I have a button that it opens a modal, then I want that when the modal is opened, the focus changes in the only input that is in the modal, the problem is that using livewire I manage to change the focus but it is an instant and the focus is removed. what I have is the following:

`<input wire:model="search" id="searcher" type="text" class="form-control" placeholder="Escriba el ciclo" autofocus="autofocus" />`
That is the input of my searches and by default it has the autofocus.

The following html is from the button that opens in a bootstrap modal with a form
`<button type="button" class="btn btn-outline-primary btn-lg" data-toggle="modal" data-target="#CreateNewCycle" wire:click="setAutofocus('cycleNew', 'searcher')">
    <i class="far fa-calendar-alt "></i> Nuevo ciclo
</button>`
In the button I have placed **wire: click = "setAutofocus ('cycleNew', 'searcher')"** so that when clicking it not only opens the modal but sends 2 parameters to a method called setAutofocus in the controller.

The method is the following:
`public function setAutofocus($nexFocus, $removeFocus)
{
   $this->emit('setAutofocus', $nexFocus, $removeFocus);
}`
In the controller it is sent by means of emit to a javascript function with the two values.

The javasscript function is as follows:
`window.livewire.on('setAutofocus', (nexFocus, removeFocus) => {
  $('#'+removeFocus).removeAttr('autofocus');
  $('#'+nexFocus).attr('autofocus', 'autofocus');
  document.getElementById(nexFocus).focus();
  console.log(nexFocus+'-'+removeFocus);
});`

This effectively focuses on the input I need and removes the autofocus from the one I had previously but it is only an instant and the focus is removed from the text field, what can I do to keep the focus, thank you very much. and disclpen my english from google.

**Comment by joshhanley** [comment]
@mldgato I think you would be better off using AlpineJs for this instead of Livewire, things like this is why Alpine was built 🙂

**Reply by mldgato** [reply]
Thank you very much, I will investigate how to use Alpine

**Reply by joshhanley** [reply]
No worries!
